### PR TITLE
Made human friendly error message on invalid sphinx queries and made $ searchable.

### PR DESCRIPTION
**Pull request by ventice**
This PR addresses the issue #1126.
Note that this fix does not return $100 for the query "\$" since $ sign is considered as part of word and one has to search it as a beginning of word to be found. On the other hand, since we have a limitation that the search string used in wildcard searches should be at least 3 characters long, "\$\*" doesn't return anything either. On the other hand "\$10\*" will return $100000 and anything else.
Standalone $ signs such as that in the sentence "In English the shorthand for buck or dollar is "$"." are found by my fix.

### CODE REVIEW

**Comment by ventice** [comment]
Technically speaking, if Sphinx on production is installed with [RE2](https://github.com/google/re2) support I can include the following directive in config to strip `$` signs off their number:
`regexp_filter           = \$(\w+) => "$ \1"`

**Comment by trang** [comment]
Alright so according to the [Sphinx documentation](http://sphinxsearch.com/docs/current/conf-regexp-filter.html):

> **12.2.64. regexp_filter**
> [...]
> Added in 2.1.1-beta. 
> [...]
> Binary packages should come with RE2 builtin. 

Currently we have in production: `Sphinx 2.2.9-id64-release (rel22-r5006)`.
And I believe it was installed from the binaries (not sure how I can verify that but I don't see any reason why we would have built it from the source), so it should have RE2 support.

File: app/views/sentences/search.ctp [hunk 1]
```diff
@@ -102,6 +103,20 @@ if ($search_disabled) {
...
 <?
+  } else {
+?>
+    <h2><?php echo __('Search error'); ?></h2>
+    <p><?php
+        echo format(
+            __(
+                'Invalid query. '.
+                'Please refer to '.
+                '<a href="{}">text search wiki</a> for more details.', true),
```
Comment (trang) [inline app/views/sentences/search.ctp]: `Please refer to text search wiki...` sounds a bit weird.
Maybe `Please refer to the search documentation...` instead.

Comment (ventice) [inline app/views/sentences/search.ctp]: Will fix.

**Comment by ventice** [comment]
Okay, I will add splitting of dollar signs off the sums in sphinx then. Looks like in this case they are not highlighted in search, so I will try to tweak it as well.

**Comment by ventice** [comment]
I failed to find a way to make sphinx highlight `$` signs adjacent to the words - `BuildExcerpts` just does not insert corresponding markers around the sign.

**Comment by trang** [comment]
> I failed to find a way to make sphinx highlight $

That's fine, it's a really minor inconvenience. We can figure out a solution later, someday.

**Pull request merged**